numpyro/contrib/hsgp/approximation.py

```diff
@@ -21,26 +21,22 @@
 import numpyro.distributions as dist
 
 
-def _non_centered_approximation(
-    phi: ArrayLike, spd: ArrayLike, m: int | list[int]
-) -> Array:
+def _non_centered_approximation(phi: Array, spd: Array, m: int) -> Array:
     with numpyro.plate("basis", m):
         beta = numpyro.sample("beta", dist.Normal(loc=0.0, scale=1.0))
 
-    return phi @ (spd * beta)
+    return jnp.asarray(phi @ (spd * beta))
```

Comment:
is this needed when inputs are Array now.

Reply:
True, changed in https://github.com/pyro-ppl/numpyro/pull/1940/commits/a7984ce697c5135086246c45692424b7603802a3